### PR TITLE
Fix djvu crash

### PR DESCRIPTION
**Pull request by hius07**
Fix crash when trying to select text in djvu (reflow off).

```
./luajit: frontend/document/koptinterface.lua:671: attempt to perform arithmetic on local 'nr_word' (a nil value)
stack traceback:
	frontend/document/koptinterface.lua:671: in function 'getTextBoxes'
	frontend/document/koptinterface.lua:1156: in function 'getTextFromPositions'
	frontend/apps/reader/modules/readerhighlight.lua:1082: in function 'handler'
	frontend/ui/widget/container/inputcontainer.lua:255: in function 'handleEvent'
	frontend/ui/uimanager.lua:1119: in function 'sendEvent'
	frontend/ui/uimanager.lua:53: in function '__default__'
	frontend/ui/uimanager.lua:1603: in function 'handleInputEvent'
	frontend/ui/uimanager.lua:1683: in function 'handleInput'
	frontend/ui/uimanager.lua:1730: in function 'run'
	./reader.lua:323: in main chunk
	[C]: at 0x00013ef9

```
The same crash for reflow mode was reported in https://github.com/koreader/koreader/issues/7850 and fixed in https://github.com/koreader/koreader/commit/e4a333a9805a1668860f0a5d134772e21e7904fb.
So here the same fix.

Unfortunately I cannot test the fix, seems my Kindle Voyage doesn't have enough memory to show the highlighted text:

```
01/07/22-09:47:24 INFO  opening file /mnt/us/books/6 TMP/Маленькие сказки про мушонка.djvu
ffi.load: libs/liblept.so.5
ffi.load: libs/libk2pdfopt.so.2
07/01/22-09:47:24 INFO  setting zoom mode to pagewidth
rendering page:0,0,4908,6382
Warning in boxaSelectBySize: boxas is empty
Warning in boxaSelectBySize: boxas is empty
Error in boxaSort2d: boxas is empty
Error in pixaSort2dByIndex: naindex not defined
Error in pixaaFlattenToPixa: paa not defined
Error in pixaGetBoxa: pixa not defined
rendering page:0,0,4908,6382
Warning in boxaSelectBySize: boxas is empty
Warning in boxaSelectBySize: boxas is empty
Error in boxaSort2d: boxas is empty
Error in pixaSort2dByIndex: naindex not defined
Error in pixaaFlattenToPixa: paa not defined
Error in pixaGetBoxa: pixa not defined
rendering page:0,0,4908,6382
Warning in boxaSelectBySize: boxas is empty
Warning in boxaSelectBySize: boxas is empty
Error in boxaSort2d: boxas is empty
Error in pixaSort2dByIndex: naindex not defined
Error in pixaaFlattenToPixa: paa not defined
Error in pixaGetBoxa: pixa not defined
rendering page:0,0,4908,6382
Warning in boxaSelectBySize: boxas is empty
Warning in boxaSelectBySize: boxas is empty
Error in boxaSort2d: boxas is empty
Error in pixaSort2dByIndex: naindex not defined
Error in pixaaFlattenToPixa: paa not defined
Error in pixaGetBoxa: pixa not defined
rendering page:0,0,4908,6382
Killed

```

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/8626)

### CODE REVIEW

**Comment by hius07** [comment]
In reflow mode the selected text is shown though.
So maybe killing is caused with some errors in Leptonica/OCR/Tesseract?
I am not familiar with this absolutely. The book is in Russian.

File: frontend/document/koptinterface.lua [hunk 1]
```diff
@@ -668,6 +668,9 @@ function KoptInterface:getNativeTextBoxesFromScratch(doc, pageno)
         local page = doc._document:openPage(pageno)
         page:getPagePix(kc)
         local boxes, nr_word = kc:getNativeWordBoxes("src", 0, 0, page_size.w, page_size.h)
+        if not boxes then
+            return
+        end
```
Comment (Frenzie) [inline frontend/document/koptinterface.lua:673]: This doesn't look quite the same as what you linked. See the page:close() and the kc:free() a few lines down.

Comment (hius07) [reply frontend/document/koptinterface.lua]: Yes, thanks, corrected. Now it does not kill but still no highlight shown
```
rendering page:0,0,4908,6382
Warning in boxaSelectBySize: boxas is empty
Warning in boxaSelectBySize: boxas is empty
Error in boxaSort2d: boxas is empty
Error in pixaSort2dByIndex: naindex not defined
Error in pixaaFlattenToPixa: paa not defined
Error in pixaGetBoxa: pixa not defined

```

**Review by Frenzie — APPROVED** [review]
lgtm in principle

**Pull request merged**